guides/common/modules/con_understanding-puppet-architecture.adoc

```diff
@@ -0,0 +1,19 @@
+[id="Understanding_puppet_architecture_{context}"]
+= Understanding Puppet Architecture
+
+Puppet uses client-server architecture with managed hosts running the Puppet agent and {Project} or any {SmartProxies} acting as a Puppet server.
+
+The Puppet agent collects _facts_ about a host and reports them to the Puppet server on each run.
+The Puppet server compares the reported _facts_ with the system state defined for the host.
+If the Puppet server finds differences (_drifts_) between the defined state and the actual state,
```

Comment:
The server always send the full definition and the Puppet agent actually does the evaluation if someting needs to be corrected to meet the definition.